v2/AdminCore.WebApp/Controllers/HolidayController.cs

```diff
@@ -22,74 +27,184 @@ namespace AdminCore.WebApi.Controllers
+    public IActionResult UpdateHoliday(UpdateHolidayViewModel updateHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<UpdateHolidayViewModel, EventDto>(updateHoliday);
+      return Ok(_holidayEventService.UpdateEvent(eventDto));
     }
 
     [HttpPut("approveHoliday")]
     public IActionResult ApproveHoliday(ApproveHolidayViewModel approveHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<ApproveHolidayViewModel, EventDto>(approveHoliday);
+      return Ok(_holidayEventService.ApproveEvent(eventDto));
     }
 
     [HttpPut("cancelHoliday")]
     public IActionResult CancelHoliday(CancelHolidayViewModel cancelHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<CancelHolidayViewModel, EventDto>(cancelHoliday);
+      return Ok(_holidayEventService.CancelEvent(eventDto));
     }
 
     [HttpPut("rejectHoliday")]
     public IActionResult RejectHoliday(RejectHolidayViewModel rejectHoliday)
     {
-      return Ok();
+      var responses = _holidayEventService.RejectEvent(
+        rejectHoliday.EventId,
+        rejectHoliday.Message,
+        int.Parse(_authenticatedUser.RetrieveUserId()));
+      return Ok(responses);
     }
 
-    [HttpGet("findByDateBetween/{eventDate}")]
-    public IActionResult GetHolidayByDateBetween(EventDate eventDate)
+    [HttpGet("findByDateBetween/{startDate}/{endDate}")]
+    public IActionResult GetHolidayByDateBetween(string startDate, string endDate)
     {
-      return Ok();
+      if (IsDatesValid(startDate, endDate))
+      {
+        return Ok(GetEventsBetweenDates(startDate, endDate));
+      }
+      return BadRequest("Please use a valid date format for start and/or end dates");
     }
 
     [HttpGet("findByHolidayStatus/{holidayStatusId}")]
-    public IActionResult GetHolidayByStatus(int holidayStatusId)
+    public IActionResult GetHolidayByStatusType(int holidayStatusId)
+    {
+      var holidaysDtos = _holidayEventService.GetByStatusType((EventStatuses)holidayStatusId, EventTypes.AnnualLeave);
+      var holidays = _mapper.Map<ResponseMessage<IList<EventDto>>, List<HolidayViewModel>>(holidaysDtos);
+
+      return Ok(holidays);
+    }
+
+    protected bool ValidateDate(string date)
+    {
+      try
+      {
+        DateTime dt = DateTime.Parse(date);
+        return true;
+      }
+      catch
+      {
+        return false;
+      }
+    }
+
+    private IActionResult CreateResponseBasedOnErrors(IList<string> errors, CreateHolidayViewModel createHolidayViewModel)
+    {
+      return errors.Any() ? CreateBadRequestResponseWithListOfErrors(errors) : CreateOkResponseWithNewEvent(createHolidayViewModel);
+    }
+
+    private IActionResult CreateOkResponseWithNewEvent(CreateHolidayViewModel createHolidayViewModel)
+    {
+      var eventDto = _mapper.Map<EventDto>(createHolidayViewModel);
+      var eventDateDto = _mapper.Map<EventDateDto>(createHolidayViewModel.DateRange);
```

Comment:
You shouldn't have to do this. Automapper should map the date ranges automatically if they have the same name in the eventDto and the mapping exists in the mapper